tools/marvin/marvin/lib/utils.py

```diff
@@ -520,4 +520,22 @@ def verifyRouterState(apiclient, routerid, allowedstates):
     if routers[0].state.lower() not in allowedstates:
         return [FAIL, "state of the router should be in %s but is %s" %
             (allowedstates, routers[0].state)]
-    return [PASS, None]
\ No newline at end of file
+    return [PASS, None]
+
+
+
+def wait_until(retry_interval=2, no_of_times=2, callback=None, *callback_args):
+    """ Utility method to try out the callback method at most no_of_times with a interval of retry_interval,
+    Will return immediately if callback returns True. The callback method should be written to return a list of values first being a boolean """
+
+    if callback is None:
+        return INVALID_INPUT
```

Comment:
This function should always return a Boolean value.  If the the ``callback`` is ``None``, why not raise a ``ValueError`` rather than return a value?  Not only is it more idiomatic Python, but it preserves the type semantics of the function.